src/modules/microdds_client/microdds_client.cpp

```diff
@@ -442,14 +574,19 @@ MicroddsClient *MicroddsClient::instantiate(int argc, char *argv[])
 	int ch;
 	const char *myoptarg = nullptr;
 
+#if defined(MICRODDS_CLIENT_UDP)
 	Transport transport = Transport::Udp;
+#else
+	Transport transport = Transport::Serial;
+#endif
 	const char *device = nullptr;
 	const char *ip = "127.0.0.1";
 	int baudrate = 921600;
 	const char *port = "15555";
 	bool localhost_only = false;
+	const char *client_namespace = nullptr;//"px4";
 
-	while ((ch = px4_getopt(argc, argv, "t:d:b:h:p:l", &myoptind, &myoptarg)) != EOF) {
+	while ((ch = px4_getopt(argc, argv, "t:d:b:h:p:l:n", &myoptind, &myoptarg)) != EOF) {
```

Comment:
If I am not wrong, the getopt considers options with letters and : are not separator but there to tell if there is an argument with the option.
So it should be` t:d:b:h:p:ln:`

Reply:
Thanks, fixed.